api/layer/object_cache_test.go

```diff
@@ -0,0 +1,170 @@
+		require.Empty(t, actual)
+	})
+
+	t.Run("startafter after all objects", func(t *testing.T) {
+		actual := trimAfterObjectName("z", objects)
+		require.Nil(t, actual)
+	})
+
+	t.Run("empty objects", func(t *testing.T) {
+		actual := trimAfterObjectName(names[0], []*ObjectInfo{})
+		require.Nil(t, actual)
+	})
+
+	t.Run("nil objects", func(t *testing.T) {
+		actual := trimAfterObjectName(names[0], nil)
+		require.Nil(t, actual)
+	})
+
+	t.Run("empty startafter", func(t *testing.T) {
+		actual := trimAfterObjectName("", objects)
+		require.Equal(t, actual, objects)
+	})
+}
+
+func TestTrimAfterObjectID(t *testing.T) {
+	var (
+		objects     []*ObjectInfo
+		ids         []*object.ID
+		numberOfIDS = 3
+	)
+
+	for i := 0; i < numberOfIDS; i++ {
+		id := randID(t)
+		objects = append(objects, &ObjectInfo{id: id})
+		ids = append(ids, id)
+	}
+
+	t.Run("existing id", func(t *testing.T) {
+		actual := trimAfterObjectID(ids[0].String(), objects)
+		require.Equal(t, objects[1:], actual)
+	})
+
+	t.Run("second to last id", func(t *testing.T) {
+		actual := trimAfterObjectID(ids[len(ids)-2].String(), objects)
+		require.Equal(t, objects[len(objects)-1:], actual)
+	})
+
+	t.Run("non-existing id", func(t *testing.T) {
+		actual := trimAfterObjectID("z", objects)
+		require.Nil(t, actual)
+	})
+
+	t.Run("last id", func(t *testing.T) {
+		actual := trimAfterObjectID(ids[len(ids)-1].String(), objects)
+		require.Nil(t, actual)
+	})
+
+	t.Run("empty id", func(t *testing.T) {
+		actual := trimAfterObjectID("", objects)
+		require.Nil(t, actual)
+	})
+}
+
+func TestObjectsListV2Cache(t *testing.T) {
+	var (
+		cacheSize = 10
+		objects   []*ObjectInfo
+		key       = "key"
+	)
+
+	for i := 0; i < cacheSize; i++ {
+		id := randID(t)
+		objects = append(objects, &ObjectInfo{id: id, Name: id.String()})
+	}
+
+	t.Run("lifetime", func(t *testing.T) {
+		cache := newListObjectsCache()
+		cache.Put(key, objects)
+
+		time.Sleep(defaultCacheLifetime - 10)
+		_, ok := cache.caches[key]
+		require.True(t, ok)
+
+		time.Sleep(time.Second * 10)
+		_, ok = cache.caches[key]
+		require.False(t, ok)
+	})
+
+	t.Run("GetByContinuationToken for valid token", func(t *testing.T) {
+		cache := newListObjectsCache()
+		cache.Put(key, objects)
+		actual := cache.GetByContinuationToken(objects[0].id.String(), key)
+		require.Equal(t, objects[1:], actual)
+	})
+
+	t.Run("GetByContinuationToken for invalid token", func(t *testing.T) {
+		cache := newListObjectsCache()
+		token := randID(t).String()
+		cache.Put(key, objects)
+		actual := cache.GetByContinuationToken(token, key)
+		require.Nil(t, actual)
+	})
+
+	t.Run("GetByMarker for valid marker", func(t *testing.T) {
+		cache := newListObjectsCache()
+		cache.Put(key, objects)
+		actual := cache.GetByMarker(objects[0].id.String(), key)
+		require.Equal(t, objects[1:], actual)
+	})
+
+	t.Run("GetByMarker for invalid token", func(t *testing.T) {
+		cache := newListObjectsCache()
+		marker := "asdf"
```

Comment:
Marker should be `zzz` for example because we have no guarantees that this key (`asdf`) will not be between existing object names

Reply:
Removed this test, it's senseless